packages/back-end/src/jobs/cacheInvalidate.ts

```diff
@@ -0,0 +1,81 @@
+import Agenda, { Job } from "agenda";
+import { AWS_CLOUDFRONT_DISTRIBUTION_ID, IS_CLOUD } from "../util/secrets";
+import AWS from "aws-sdk";
+import { CreateInvalidationRequest } from "aws-sdk/clients/cloudfront";
+import { ExperimentInterface } from "../../types/experiment";
+import { getAllApiKeysByOrganization } from "../services/apiKey";
+
+const INVALIDATE_JOB_NAME = "fireInvalidate";
+type InvalidateJob = Job<{
+  url: string;
+}>;
+
+let agenda: Agenda;
+export default function (ag: Agenda) {
+  agenda = ag;
+
+  // Fire webhooks
+  agenda.define(INVALIDATE_JOB_NAME, async (job: InvalidateJob) => {
+    const { url } = job.attrs.data;
+
+    if (!url || url === "") return;
+
+    // we should eventually restructure this to a cron job that collects
+    // a full list of all URLs to be cleared, and send one clear request
+    // to AWS, which might help reduce the number of calls to aws.
+    if (AWS_CLOUDFRONT_DISTRIBUTION_ID) {
```

Comment:
If there is no distribution id, the job should never be queued in the first place.

Reply:
That's correct, the job is not queued if this is missing. This is just a sanity check.